### PR TITLE
Updates to defer/stream RFC

### PR DESCRIPTION
**Pull request by robrichard**
This builds on top of @Keweiqu's changes in https://github.com/graphql/graphql-spec/pull/679

### CODE REVIEW

**Comment by mike-marcacci** [comment]
This is just a nit, but for the OSS RFC, we _probably_ want to camelCase the identifiers `initial_count` and `is_final` to match the convention used elsewhere in GraphQL.

File: rfcs/DeferStream.md [hunk 1]
```diff
@@ -3,18 +3,146 @@ RFC: GraphQL Defer and Stream Directives
...
+  * When `true` fragment may be deferred, if omitted defaults to `true`.
+* `label: String!`
+  * A unique label across all `@defer` and `@stream` directives in an operation.
+  * This `label` should be used by GraphQL clients to identify the data from patch responses and associate it with the correct fragment.
+
+## `@stream`
+
+The `@stream` directive may be provided for a field of `List` type so that the backend can leverage technology such asynchronous iterators to provide a partial list in the initial response, and additional list items in subsequent responses. `@include` and `@skip` take presedence over `@stream`.
+
+### `@stream `arguments
```
Comment (fotoetienne) [inline rfcs/DeferStream.md:50]: Should there also be a way for clients to limit the number items that would be returned in a stream? i.e. `maxCount`

File: rfcs/DeferStream.md [hunk 2]
```diff
@@ -3,18 +3,146 @@ RFC: GraphQL Defer and Stream Directives
...
+## Splitting queries
 
+Given a query where some fields are expensive and non-essential, non-essential fields can be fetched in a separate query issued after an initial query. In the context of lists, this may mean fetching only the first few items in a list and issuing a follow-up pagination request for additional items. By separating the requests based on data priority, we have achieved a faster delivery time to essential data. However, query splitting imposes several **trade-offs** on applications:
+
+**Increased latency for lower priority fields,** as now these fields have to wait for the original query to complete and have added network round trips
+**Client resource contention.** Issuing multiple requests can increase contention on scarce resources like battery and antenna.
```
Comment (fotoetienne) [inline rfcs/DeferStream.md:22]: Would this be true for requests that are multiplexed over HTTP/2 connections?

File: rfcs/DeferStream.md (hunk 1 above)
Comment (robrichard) [reply rfcs/DeferStream.md:53]: @fotoetienne is this different from the pagination arguments that would typically be used on list or connection fields? Would `maxCount` affect the streaming mechanism itself or would it change the data that is returned?

File: rfcs/DeferStream.md (hunk 2 above)
Comment (robrichard) [reply rfcs/DeferStream.md:22]: I'm not sure. Splitting queries usually requires data from the first query to be used as input to the second query, so they are not able to run in parallel.

**Comment by leebyron** [comment]
Including these updates to keep the RFC moving, thank you!

**Pull request merged**